File: VB/MVCxDashboardViewer_PerformDrillDown/Controllers/HomeController.vb
Imports DevExpress.DashboardWeb.Mvc
Imports DevExpress.DataAccess.ConnectionParameters
Imports System.Web.Mvc

Namespace MVCxDashboardViewer_PerformDrillDown.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        <ValidateInput(False)> _
        Public Function DashboardViewerPartial() As ActionResult
            Return PartialView("_DashboardViewerPartial", DashboardViewerSettings.Model)
        End Function
        Public Function DashboardViewerPartialExport() As FileStreamResult
            Return DashboardViewerExtension.Export("DashboardViewer", DashboardViewerSettings.Model)
        End Function
    End Class
    Friend Class DashboardViewerSettings
        Public Shared ReadOnly Property Model() As DashboardSourceModel
            Get

                Dim model_Renamed As New DashboardSourceModel()
                model_Renamed.DashboardSource = System.Web.Hosting.HostingEnvironment.MapPath("~\App_Data\Dashboard.xml")

                model_Renamed.ConfigureDataConnection = Sub(sender, e)
                    If e.ConnectionName = "nwindConnection" Then
                        Dim parameters As Access97ConnectionParameters = CType(e.ConnectionParameters, Access97ConnectionParameters)
                        Dim databasePath As String = System.Web.Hosting.HostingEnvironment.MapPath("~\App_Data\nwind.mdb")
                        parameters.FileName = databasePath
                    End If
                End Sub
                Return model_Renamed
            End Get
        End Property
    End Class
End Namespace

File: CS/MVCxDashboardViewer_PerformDrillDown/Controllers/HomeController.cs
using DevExpress.DashboardWeb.Mvc;
using DevExpress.DataAccess.ConnectionParameters;
using System.Web.Mvc;

namespace MVCxDashboardViewer_PerformDrillDown.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        [ValidateInput(false)]
        public ActionResult DashboardViewerPartial()
        {
            return PartialView("_DashboardViewerPartial", DashboardViewerSettings.Model);
        }
        public FileStreamResult DashboardViewerPartialExport()
        {
            return DashboardViewerExtension.Export("DashboardViewer", DashboardViewerSettings.Model);
        }
    }
    class DashboardViewerSettings
    {
        public static DashboardSourceModel Model
        {
            get
            {
                DashboardSourceModel model = new DashboardSourceModel();
                model.DashboardSource = System.Web.Hosting.HostingEnvironment.MapPath(@"~\App_Data\Dashboard.xml");

                model.ConfigureDataConnection = (sender, e) => {
                    if (e.ConnectionName == "nwindConnection")
                    {
                        Access97ConnectionParameters parameters = (Access97ConnectionParameters)e.ConnectionParameters;
                        string databasePath = System.Web.Hosting.HostingEnvironment.MapPath(@"~\App_Data\nwind.mdb");
                        parameters.FileName = databasePath;
                    }
                };
                return model;
            }
        }
    }
}